shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareExecutor.java

```diff
@@ -66,9 +73,41 @@
     }
     
     private int getProjectionCount(final SQLStatement sqlStatement) {
-        return sqlStatement instanceof SelectStatement ? ((SelectStatement) sqlStatement).getProjections().getProjections().size() : 0;
+        return sqlStatement instanceof SelectStatement
+                ? getProjectionCount((SelectStatement) sqlStatement) : 0;
```

Comment:
Hi could you move this relevant handling to `ProjectionsContext` which is a suitable place for this info.

Reply:
I'm sorry, I read the Proxy code for the first time, and I don’t understand how to move it.

Reply:
Can it be moved to ProjectionsContext static api?

Reply:
Hi, sorry for my late response. I mean could we do some improvement on this bug-fix?
Please look at `getColumnCount()` of `ShardingSphereResultSetMetaData`. Actually it does a similar thing with you. So can we leverage the `ProjectionsContext` to deal with this issue? 
Maybe first we can call `SQLStatementContextFactory` to get `SelectStatementContext` and `ProjectionsContext`.
Then, could you figure out whether `ProjectionsContext` can return the `actual projection count`?

BTW, the failed CI still needs your attention. Thanks, hope this PR merged ASAP.

Reply:
ok, thanks!